app/scripts/controllers/createSecret.js

```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:CreateSecretController
+ * @description
+ * # CreateSecretController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('CreateSecretController', function ($scope, $filter, DataService, AlertMessageService) {
+
+    $scope.newSecret = {
+      type: $scope.type === "Git Secret" ? "Source Secret" : $scope.type,
+      authType: ($scope.type === "Source Secret" || $scope.type === "Git Secret") ? "Basic Authentication" : "Credentials",
+      data: {},
+      linkSecret: false,
+      serviceAccountToLink: "",
```

Comment:
Do we need both the boolean and the service account property? If service account is not empty, we know to link.